File: bmh_parent/bmh_rope_web2/src/main/webapp/js/service/systemPermissionService.js
//服务层
app.service('systemPermissionService',function($http){


	this.updateStatus=function(status,systemPermissionId){
		return $http.get('../../systemPermission/updateStatus/'+status+"/"+systemPermissionId);
	}
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../../systemPermission/findAll');
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../../systemPermission/findPage.do?page='+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../../systemPermission/findOne/'+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../../systemPermission/add',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../../systemPermission/update.do',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../../systemPermission/delete.do?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../../systemPermission/search?page='+page+"&rows="+rows, searchEntity);
	}    	
});

File: bmh_parent/bmh_rope_interface/src/main/java/com/bmh/rope/service/SystemPermissionService.java
package com.bmh.rope.service;

import com.bmh.pojo.Permission;
import entity.PageResult;

import java.util.List;

/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface SystemPermissionService {

    /**
     * 返回全部列表
     * @return
     */
    public List<Permission> findAll();


    /**
     * 返回分页列表
     * @return
     */
    public PageResult findPage(int pageNum, int pageSize);


    /**
     * 增加
     */
    public void add(Permission systemPermission);


    /**
     * 修改
     */
    public void update(Permission systemPermission);


    /**
     * 根据ID获取实体
     * @param id
     * @return
     */
    public Permission findOne(String id);


    /**
     * 批量删除
     * @param ids
     */
    public void delete(String[] ids);

    /**
     * 分页
     * @param pageNum 当前页 码
     * @param pageSize 每页记录数
     * @return
     */
    public PageResult findPage(Permission systemPermission, int pageNum, int pageSize);

    void updateStatus(String status, String systemPermissionId);

    Permission findByUserName(String username);
}
